lib/max_subarray.py

```diff
@@ -2,11 +2,23 @@
 def max_sub_array(nums):
     """ Returns the max subarray of the given list of numbers.
         Returns 0 if  nums is None or an empty list.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(1)
     """
     if nums == None:
         return 0
     if len(nums) == 0:
         return 0
-    pass
+
+    max = - 10000000000000000000000000000000000000000000
```

Comment:
Another approach would be to initialize the maximum to some value actually found in the list, say `nums[0]`, which we know must at least exist from the guard checks.